### ISSUE TITLE
ssl cert weirdness

### ISSUE THREAD

**Issue by gillyspy** [terreng/simple-web-server#39]
I was unable to get the 1.0 build to work with SSL.  I built from https://github.com/terreng/simple-web-server/commit/9a76b8b90f7bcacb4039576cc9cd359c09e5b45d and still no joy. 
I noticed the key/cert info was getting parsed incorrectly from config.json

adding this transformation worked as a band-aid for me: (index.js):
```
		let [key,cert] = ( ({httpsKey,httpsCert})=>{ return [httpsKey,httpsCert]})(serverconfig); 
		[key,cert] = [key,cert].map( value=>{
                      return value
			.replaceAll(/(?<!(BEGIN|RSA|PRIVATE|END))[ ]/g,'\n')
			.replace(/-[^-]+$/,'');
                 });	    
		var server = https.createServer({key, cert}); 
               // var server = https.createServer({key: serverconfig.httpsKey, cert: serverconfig.httpsCert});
```

(i don't think it's the right way to solve this so I didn't create a PR)

**Comment by terreng** [comment]
@ethanaobrien Please take a look

**Comment by ethanaobrien** [comment]
@gillyspy Were there any errors in the console?

**Comment by terreng** [comment]
Thanks for opening the issue! I assume you were using a custom cert, not an auto-generated one, is that right?

**Comment by gillyspy** [comment]
Pretty sure I was having the issue with auto generated as well so I tried
my own and then finally resolved to download source and "fix" the issue.  I
think the root issue was the parsing for storage and then parsing for use
is not consistent. Ie you manually put in a key with line break sequence
but then it saves it in config with spaces .. so next time you load it,
it's broken

**Comment by ethanaobrien** [comment]
@gillyspy Is it possible that you would be able to do some debugging to find exactly where the parse error is. I've used ssl on about 5 different machines, 3 windows, 2 linux, and I havent seen this problem at all. I can't fix it unless I can find whats causing it.

Thanks!

**Comment by terreng** [comment]
@ethanaobrien Please take a look at the code fix provided and see if you can identify what problem that regex is fixing. When parsing, could you just remove the beginning and ending lines and line breaks (all non base64 characters?) so that it works if it's entered different ways into the config file?

**Comment by ethanaobrien** [comment]
Okie. Ill do some looking into

**Comment by gillyspy** [comment]
i took a quick look and this might not be a bug so much as it is an enhancement.  The chrome webserver extension that you guys "replaced" allowed keys with a space ` ` as the delimiter but you are using `\r\n`. 

The change I gave you will effectively allow both.  What I probably did was copied the key incorrectly -- maybe even from the old config.

**Comment by terreng** [comment]
@gillyspy Thanks so much for the additional information. We really appreciate you bringing this to our attention.

@ethanaobrien Let's support both space and newline delimiters for certs.

**Comment by ethanaobrien** [comment]
@terreng So do you want me to just replace all spaces with a newline delimiter, correct?

**Comment by terreng** [comment]
> @terreng So do you want me to just replace all spaces with a newline delimiter, correct?

Yeah, if you think that's the best way to do it.

**Comment by ethanaobrien** [comment]
@terreng Just added to my pr

**Comment by terreng** [comment]
Great! Add "Closes #39" to your pr.

Thanks again for your report, gillyspy!
